test/Test.cpp

```diff
@@ -120,41 +176,50 @@ int main(int,char**)
 	java::lang::Object object = java::lang::Integer(23754);
 	if (jni::InstanceOf<java::lang::Number>(object) && jni::Cast<java::lang::Number>(object))
 	{
-		printf("%d\n", static_cast<int>(java::lang::Number(object)));
+		printf("%d\n", static_cast<jint>(java::lang::Number(object)));
```

Comment:
Why this change? %d specifier means int and it's not guaranteed that jint is typedefed to int.

Reply:
java::lang::Number doesn't have operator for casting to int
```
struct Number : ::java::lang::Object
{
	static jni::Class __CLASS;

	operator ::java::io::Serializable();
	::jbyte ByteValue() const;
	::jdouble DoubleValue() const;
	::jfloat FloatValue() const;
	::jint IntValue() const;
	::jlong LongValue() const;
	::jshort ShortValue() const;
	static jobject __Constructor();
	Number() : ::java::lang::Object(__Constructor()) { __Initialize(); }
	explicit Number(jobject o) : ::java::lang::Object(o) { __Initialize(); }
	Number(const Number& o)  : ::java::lang::Object(o) { __Initialize(); }
Number(Number&& o) = default;
Number& operator=(const Number& o) = default;
Number& operator=(Number&& o) = default;
operator ::jbyte() const;
operator ::jshort() const;
operator ::jint() const;
operator ::jlong() const;
operator ::jfloat() const;
operator ::jdouble() const;
private:
	void __Initialize();

};
```

thus you would get
1>E:\Projects\android-jni-bridge\test\Test.cpp(179): error C2440: 'static_cast': cannot convert from 'java::lang::Number' to 'int'

Reply:
I think it's very good that MS complains about it, since jint even though under hood the same, is declared as new type.

So, can you explain, if you want something to fix here, or leave it as it is ?

Reply:
Checked our OpenJDK and on Mac jint is typedefed as int, while on Windows it's typedefed as long.
So you need to either double cast or cast to something universal (maybe jlong is universal?).

Reply:
Fixed it, casting to jlong now